### ISSUE TITLE
Bug in circuit_timing.hpp

### ISSUE THREAD

**Issue by mattdouthwaite** [HPCE/hpce-2015-cw5#5]
As discussed after the lecture today. There appears to be a bug in either the Delay function or ReferenceExecute. The loop that calls the delay function (line 130) iterates i from 0 to less than the number of flip flops, then passes i into delay. The first if statement in Delay then checks if i is less than the number of flip flops, which will always be true as that is the iteration space.
So the programme will never enter the 'else' part of Delay.
Agreed as probably a bug and fix will come soon.

**Comment by m8pple** [comment]
Yup, my bad - it's a single line error from when I copied it over
from my reference. The bug is on line 129 of include/puzzler/puzzles/circuit_timing.hpp,
which reads:

```
unsigned got=delay(i, pInput);
```

when it _should_ say:

```
unsigned got=delay(pInput->wires.at(i), pInput);
```

At some point I moved the indirection out of the delay function, but joined
it with a different version of the caller. With the modification it does actual
circuit timing.

I will push a change and then email all.
